api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -142,67 +122,18 @@
 	 * @should fetch users assigned given role
 	 * @should not fetch user that does not belong to given role
 	 */
-	@Authorized( { PrivilegeConstants.VIEW_USERS })
+	@Authorized( { PrivilegeConstants.GET_USERS })
 	public List<User> getUsersByRole(Role role) throws APIException;
 	
 	/**
-	 * Save changes to given <code>user</code> to the database.
-	 * 
-	 * @param user
-	 * @throws APIException
-	 * @see #saveUser(User, String)
-	 * @deprecated replaced by {@link #saveUser(User, String)}
-	 */
-	@Deprecated
-	@Authorized( { PrivilegeConstants.EDIT_USERS })
-	public void updateUser(User user) throws APIException;
-	
-	/**
-	 * Use should be UserService.saveUser(user.addRole(role))
-	 * 
-	 * @deprecated use {@link org.openmrs.User#addRole(Role)}
-	 */
-	@Deprecated
-	@Authorized( { PrivilegeConstants.EDIT_USERS })
-	public void grantUserRole(User user, Role role) throws APIException;
-	
-	/**
-	 * Use UserService.saveUser(user.removeRole(role))
-	 * 
-	 * @deprecated use {@link org.openmrs.User#removeRole(Role)}
-	 */
-	@Deprecated
-	@Authorized( { PrivilegeConstants.EDIT_USERS })
-	public void revokeUserRole(User user, Role role) throws APIException;
-	
-	/**
-	 * Mark user as voided (effectively deleting user without removing their data &mdash; since
-	 * anything the user touched in the database will still have their internal identifier and point
-	 * to the voided user for historical tracking purposes.
-	 * 
-	 * @param user
-	 * @param reason
-	 * @return the given user voided out
-	 * @throws APIException
-	 * @should void user and set attributes
-	 * @deprecated use {@link #retireUser(User, String)}
-	 */
-	@Deprecated
-	@Authorized( { PrivilegeConstants.EDIT_USERS })
-	public User voidUser(User user, String reason) throws APIException;
-	
-	/**
-	 * Clear voided flag for user (equivalent to an "undelete" or Lazarus Effect for user)
+	 * Updates a given <code>user</code> in the database.
 	 * 
 	 * @param user
-	 * @return the given user unvoided
+	 * @return the saved user
 	 * @throws APIException
-	 * @should unvoid and unmark all attributes
-	 * @deprecated use {@link #unretireUser(User)}
 	 */
-	@Deprecated
 	@Authorized( { PrivilegeConstants.EDIT_USERS })
-	public User unvoidUser(User user) throws APIException;
+	public User saveUser(User user) throws APIException;
```

Comment:
I thought createUser was deprecated and removed, saveUser should take an optional password so that we don't need a new saveUser, did i miss the design call where this was decided?